rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,206 @@
+# Stop processing transactions / stop the network after a certain slot
+
+This PR describes the feature to stop processing transactions and to stop the
+network after a certain slot, to be used in the Berkeley hard fork.
+
+## Summary
+
+Transactions come from a client or the gossip network and are processed by BPs
+and SNARK workers to be included in blocks. These blocks are propagated through
+the network and validated by other participants.
+
+In this RFC, the procedure to stop processing any new transactions and to
+stop the network after a certain slot is described. This is, we define two
+slots: the first one is the slot after which any blocks produced will include no
+transaction at all and no fee payments, and the second one is the slot after
+which no blocks are produced and blocks received are rejected.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. This new genesis ledger should be
+produced from a stabilised staged ledger from the preceding network. This is, we
+define a point in time (slot) where the network continues to operate but with no
+"activity". In detail, after this slot, the network continues to produce blocks
+but without including any transactions, and ensuring any fees paid are 0. This
+will run for a certain number of slots, after which the network will stop
+producing blocks. This will allow the network to stabilise and produce a new
+genesis ledger from the last ledger produced by the network.
+
+This feature enables part of this procedure, by adding the definition of the
+slots and the mechanisms to stop the node from processing transactions and to
+stop the networks after those slots.
+
+## Detailed design
+
+The procedure to stop processing transactions and producing/validating empty
+blocks after a certain slot will be as follows:
+
+* There will be a configuration parameter set at compile-time that will define
+  the slot at which the node will stop processing transactions.
+* The previous configuration should be overridable at runtime by optional CLI
```

Comment:
We need to be careful with runtime overrides of this. Imagine a scenario where an adversary provides a malicious config. The daemon software is the same, and the chain id is the same (since this config doesn't and shouldn't contribute to chain id), but daemon's running with this config will all fork from the network at a predetermined height. I'd recommend not letting this be overridable at runtime in order to avoid this.

Reply:
You are suggesting not having any CLI flags to change this feature and letting it only be defined at compile time. Right?

Reply:
I have replaced the bullet with your suggestion and removed every mention of configuration via CLI flags.